### PR TITLE
Make Graql tests run on all backends

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

**Comment by aelred** [comment]
Please don't accept this until it passes Jenkins!

**Comment by sheldonkhall** [comment]
please retest this please

**Comment by sheldonkhall** [comment]
please retest this please

**Comment by sheldonkhall** [comment]
please retest this please

**Comment by aelred** [comment]
please retest this please

**Comment by aelred** [comment]
OH MY GOD FINALLY

File: mindmaps-test/src/test/java/io/mindmaps/test/AbstractEngineTest.java [hunk 1]
```diff
@@ -21,25 +21,21 @@
...
-
-    protected static MindmapsGraphFactory factory;
-    protected static MindmapsGraph graph;
-
-    private static AtomicBoolean ENGINE_ON = new AtomicBoolean(false);
-
+/**
+ * Abstract test class that automatically starts the backend and engine and provides a method to get a graph factory
+ */
+public class AbstractEngineTest {
```
Comment (sheldonkhall) [inline mindmaps-test/src/test/java/io/mindmaps/test/AbstractEngineTest.java]: this should be abstract

File: mindmaps-test/src/test/java/io/mindmaps/test/graql/reasoner/inference/AbstractInferenceTest.java [hunk 2]
```diff
@@ -32,7 +33,7 @@
 import static org.junit.Assert.assertEquals;
 
 
-public class AbstractInferenceTest {
+public class AbstractInferenceTest extends AbstractEngineTest {
```
Comment (sheldonkhall) [inline mindmaps-test/src/test/java/io/mindmaps/test/graql/reasoner/inference/AbstractInferenceTest.java]: Either remove this or figure out what AbstractReasonerEngineTest looks like

**Pull request merged**